controllers/update_pods_test.go

```diff
@@ -246,5 +249,70 @@ var _ = Describe("update_pods", func() {
 				Expect(updates).To(HaveLen(1))
 			})
 		})
+
+		When("max unavailable pods is set with a percent value and there are process groups with pods in pending status", func() {
+			BeforeEach(func() {
+				expectedError = true
+				cluster.Spec.MaxUnavailablePods = intstr.FromString("1%")
+				Expect(k8sClient.Update(context.TODO(), cluster)).NotTo(HaveOccurred())
+
+				var numPendingPods int
+				for _, processGroup := range cluster.Status.ProcessGroups {
+					if processGroup.ProcessClass.IsStateful() {
+						processGroup.ProcessGroupConditions = append(processGroup.ProcessGroupConditions, fdbv1beta2.NewProcessGroupCondition(fdbv1beta2.PodPending))
+						numPendingPods++
+						if numPendingPods == 8 {
```

Comment:
This case is never hit right? The initial value of `numPendingPods` will be 0 and before we do the check here we are increasing the value by 1, so it will always be > 0 and all process groups will get this condition? Was the idea of `numPendingPods` to be used for debugging?

Reply:
The idea here was to set the PodPending condition for a certain number (8 in this case) of process groups. Once the desired number is reached `numPendingPods` we skip setting the PodsPending condition for the remaining process groups.

Does that make sense?

Reply:
That makes sense, I just was confused what the benefit of the previous check was.